source/documentation/tools/create-a-derived-table/project-structure.md

```diff
@@ -1,66 +1,723 @@
+        cast(credit_amount / 100.0) as float) as credit_amount,
+        cast(total_amount / 100.0) as float) as total_amount,
+        ---------- booleans
+        is_recoverable,
+
+        ---------- dates
+        date(paid_date) as paid_date,
+
+        ---------- timestamps
+        transction_at
+
+    from source
+
+)
+
+select * from renamed
+```
+
+- Based on the above, the most standard types of staging model transformations are:
+  - ✅ **Renaming**
+  - ✅ **Type casting**
+  - ✅ **Basic computations** (e.g. cents to dollars)
+  - ✅ **Categorizing** (using conditional logic to group values into buckets or booleans, such as in the `case when` statements above)
+  - ✅ **Generating keys** Create unique surrogate keys using the dbt utils function `dbt_utils.generate_surrogate_key(`
+  - ❌ **Joins** — the goal of staging models is to clean and prepare individual source-conformed concepts for downstream usage. We're creating the most useful version of a source system table, which we can use as a new modular component for our project. In our experience, joins are almost always a bad idea here — they create immediate duplicated computation and confusing relationships that ripple downstream — there are occasionally exceptions though.
+  - ❌ **Aggregations** — aggregations entail grouping, and we're not doing that at this stage. Remember - staging models are your place to create the building blocks you’ll use all throughout the rest of your project — if we start changing the grain of our tables by grouping in this layer, we’ll lose access to source data that we’ll likely need at some point. We just want to get our individual concepts cleaned and ready for use, and will handle aggregating values downstream.
+- ✅ **Materialised as views.** Looking at a partial view of our `dbt_project.yml` below, we can see that we’ve configured the entire staging directory to be materialised as <Term id='view'>views</Term>. As they’re not intended to be final artefacts themselves, but rather building blocks for later models, staging models should typically be materialised as views for two key reasons:
+
+  - Any downstream model (discussed more in [marts](/best-practices/how-we-structure/4-marts)) referencing our staging models will always get the freshest data possible from all the component views it’s pulling together and materialising
+  - It avoids wasting space in the warehouse on models that are not intended to be queried by data consumers, and thus do not need to perform as quickly or efficiently
+
+    ```yaml
+    # dbt_project.yml
+    mojap_derived_talbes:
+      models:
+          staging:
+            +materialised: view
+    ```
+
+**:::TIP:** During development of a data model, it may be useful to materialise all your models as tables. This will allow you to more easily debug issues. When you are ready to merge your work with main then you can change the materialisation back to view. **:::**
```

Comment:
https://www.freecodecamp.org/news/how-to-create-notice-blocks-in-markdown/
could be useful here